### PR TITLE
Track errors, exceptions and network requests automatically with custom properties in Console app 

### PR DESCRIPTION
**Pull request by pavinduLakshan**
### Purpose
Note $subject

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/core/utils/http-utils.ts [hunk 1]
```diff
@@ -69,17 +79,16 @@ export class HttpUtils {
...
-            error.response.data &&
-            error.response.data.code
-        ) {
-            EventPublisher.getInstance().publish("console-error-http-request-error", {
-                "code": error.response.data.code,
-                "status": error.response.status ? error.response.status as number : "",
-                "type": "error-response"
-            });
-        }
+        const currentTimeStamp = new Date().getTime()
```
Comment (brionmario) [inline apps/console/src/features/core/utils/http-utils.ts:82]: Let's add types for these.

File: apps/console/src/features/core/utils/http-utils.ts [hunk 2]
```diff
@@ -49,8 +50,17 @@ export class HttpUtils {
     /**
      * Callback to be fired on every Http request success.
      */
-    public static onHttpRequestSuccess(): void {
+    public static onHttpRequestSuccess(response: HttpResponse): void {
         // TODO: Handle any conditions required on request success.
+        const timeStamp = new Date().getTime()
```
Comment (brionmario) [inline apps/console/src/features/core/utils/http-utils.ts:55]: Let's add types for these.

File: apps/console/src/features/core/utils/event-publisher.ts [hunk 3]
```diff
@@ -95,6 +95,34 @@ export class EventPublisher {
...
+                );
+        }
+
+        analyticsConfig.EventPublisherExtension.record &&
+            analyticsConfig.EventPublisherExtension.record(
+                pathname, startTime, duration, responseCode, isSuccess,
+            );
+
+        return;
+
```
Comment (brionmario) [inline apps/console/src/features/core/utils/event-publisher.ts:123]:
```suggestion
```

File: apps/console/src/features/core/utils/http-utils.ts (hunk 1 above)
Comment (pavinduLakshan) [reply apps/console/src/features/core/utils/http-utils.ts]: done

File: apps/console/src/features/core/utils/http-utils.ts (hunk 2 above)
Comment (pavinduLakshan) [reply apps/console/src/features/core/utils/http-utils.ts]: done

File: features/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2 [hunk 4]
```diff
@@ -23,6 +23,13 @@
         {% endfor %}
         {% endif %}
     },
+    "analytics": {
+        {% if console.analytics.metrics is defined %}
+        {% for param, value in console.analytics.metrics.items() %}
```
Comment (brionmario) [inline features/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2:28]: If `console.analytics.metrics is defined` we need to add `metricts` object right?
```suggestion
       "metricts" {
        {% for param, value in console.analytics.metrics.items() %}
```

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/core/utils/http-utils.ts (hunk 1 above)
Comment (pavinduLakshan) [reply apps/console/src/features/core/utils/http-utils.ts]: fixed with 2ea3951

File: apps/console/src/features/core/utils/http-utils.ts (hunk 2 above)
Comment (pavinduLakshan) [reply apps/console/src/features/core/utils/http-utils.ts]: fixed with 2ea3951

File: apps/console/src/features/core/utils/event-publisher.ts (hunk 3 above)
Comment (pavinduLakshan) [reply apps/console/src/features/core/utils/event-publisher.ts:123]: fixed with 2ea3951

File: features/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2 (hunk 4 above)
Comment (pavinduLakshan) [reply features/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2:29]: fixed with 52d1c5b

File: apps/console/src/features/core/utils/event-publisher.ts [hunk 5]
```diff
@@ -95,6 +95,34 @@ export class EventPublisher {
...
+        pathname: string, startTimeInMs: number, duration: number, responseCode: number, isSuccess: boolean,
+        customProperties?: {
+            [key: string]: string | Record<string, unknown> |
+            number
+        }): void {
+
+        /**
+         * If you want to do any event logging, do it here.
+         * custom properties are passed here.
+        */
```
Comment (NipuniBhagya) [inline apps/console/src/features/core/utils/event-publisher.ts:108]: Fix the indentation.

**Pull request merged**